sql-plugin/src/main/scala/org/apache/spark/sql/rapids/stringFunctions.scala

```diff
@@ -487,6 +487,60 @@ case class GpuInitCap(child: Expression) extends GpuUnaryExpression with Implici
     }
 }
 
+case class GpuStringRepeat(input: Expression, repeatTimes: Expression)
+    extends GpuBinaryExpression with ImplicitCastInputTypes with NullIntolerant {
+  override def left: Expression = input
+  override def right: Expression = repeatTimes
+  override def dataType: DataType = input.dataType
+  override def inputTypes: Seq[AbstractDataType] = Seq(StringType, IntegerType)
+
+  def doColumnar(lhs: GpuScalar, rhs: GpuColumnVector): ColumnVector =
+    throw new UnsupportedOperationException(s"Cannot columnar evaluate expression: $this")
```

Comment:
Why does this need to be the case?  If the input string is a scalar but the repeat times are columnar, we can fill a column of the same length with the scalar and make this work, right?  For example, something like this:
```
  withResource(GpuColumnVector.from(lhs, rhs.getRowCount, lhs.dataType)) { replicatedInput =>
    doColumnar(replicatedInput, rhs)
  }
```

It looks like Spark CPU supports this case, e.g.:
```
scala> val df = Seq(1, 2, 3).toDF
df: org.apache.spark.sql.DataFrame = [value: int]

scala> df.withColumn("repeated", expr("repeat('abc', value)")).show
+-----+---------+
|value| repeated|
+-----+---------+
|    1|      abc|
|    2|   abcabc|
|    3|abcabcabc|
+-----+---------+
```